api/v1alpha1/bundle_types.go

```diff
@@ -18,48 +18,73 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	runtime "k8s.io/apimachinery/pkg/runtime"
 )
 
-// Source locates content to stage.
-type Source string
+type ProvisionerID string
+type BundleConditionType string
+
+const (
+	TypeUnpacked = "Unpacked"
+
+	ReasonUnpackPending    = "UnpackPending"
+	ReasonUnpacking        = "Unpacking"
+	ReasonUnpackSuccessful = "UnpackSuccessful"
+	ReasonUnpackFailed     = "UnpackFailed"
+
+	PhasePending   = "Pending"
+	PhaseUnpacking = "Unpacking"
+	PhaseFailing   = "Failing"
+	PhaseUnpacked  = "Unpacked"
+)
 
 // BundleSpec defines the desired state of Bundle
 type BundleSpec struct {
-	// Class specifies the name of the provisioner that should manage the Bundle.
-	// +optional
-	Class string `json:"class,omitempty"`
+	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
+	ProvisionerClassName string `json:"provisionerClassName"`
 
-	// Source locates all remote content to be staged by the Bundle.
-	Source Source `json:"source"`
-}
+	// Image is the bundle image that backs the content of this bundle.
+	Image string `json:"image"`
 
-// Content surfaces staged content for clients to access.
-type Content struct {
-	runtime.RawExtension `json:"-"`
+	// ImagePullSecrets is a list of pull secrets to have available to
+	// pull the referenced image.
+	ImagePullSecrets []ImagePullSecret `json:"imagePullSecrets,omitempty"`
 }
 
-type ProvisionerID string
+type ImagePullSecret struct {
```

Comment:
As discussed in our conversation this is not really secure: it allows the provisioner to potentially copy secrets from any namespace into it's own namespace. Maybe we should just drop support for private bundle images for now?

Reply:
Good catch - removed this in 3aec843fd8fe1f0f73d847869d6447d960e81b62